### PR TITLE
Feature update to sq9

### LINKED ISSUE #66: Update to SonarQube LTS 9

### PR DESCRIPTION
**Pull request by diegorodriguez31**
## Proposed changes

Update new tools on last version compatible with SonarQube LTS 9
Fix some testing bugs

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Issues closed by changes

- [x] Close #66

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I agree with the [CODE OF CONDUCT](CODE_OF_CONDUCT.md)
- [x] Lint and unit tests pass locally with my changes
- [x] Codacy validates the quality of the modifications
- [x] GitHub Actions CI workflow passes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -63,14 +63,6 @@ jobs:
           cd tests/
           python3 -m pip install -r requirements.txt
           python3 -m pytest -v |& tee -a ../tests_logs.txt
-      # Publish test results in the pull request discussion
```
Comment (louisjdmartin) [inline .github/workflows/ci.yml:66]: Pourquoi ne plus afficher les logs dans les commentaires de la PR ?

File: Dockerfile [hunk 2]
```diff
@@ -1,103 +1,101 @@
-FROM sonarqube:7.9.4-community AS framac
+# This image is based on a LTS version of SonarQube
+FROM sonarqube:9.9.1-community
 
-## ====================== INSTALL FRAMA-C =============================
+LABEL maintainer="CATLab"
```
Comment (louisjdmartin) [inline Dockerfile:4]:
```suggestion
LABEL maintainer="CNES CAT Lab"
```

File: Dockerfile [hunk 3]
```diff
@@ -1,103 +1,101 @@
...
 ENV PATH /usr/local/bin:${PATH}
 
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends gpg dirmngr \
+    && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 648ACFD622F3D138 \
+    && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 0E98404D386FA1D9 \
+    && apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 605C66F00D6C9793 \
+    && rm -rf /var/lib/apt/lists/*
+
 RUN echo 'deb http://ftp.fr.debian.org/debian/ bullseye main contrib non-free' >> /etc/apt/sources.list \
```
Comment (louisjdmartin) [inline Dockerfile:81]: Pourquoi enlever le numéro de version de GCC ?
Le Warning de Codacy est plutot pertinent si on veux garder une image reproductible

**Review by louisjdmartin — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (diegorodriguez31) [reply .github/workflows/ci.yml:66]: Ce plugin ne fonctionne plus dans notre cas, c'est la même notion que sur l'image SonarQube. On avait décidé de ne plus l'utiliser, et sans le remplacer.

**Pull request merged**